algolia/transport/transport.go

```diff
@@ -140,29 +144,56 @@ func (t *Transport) Request(
 	return errs.ErrNoMoreHostToTry
 }
 
+func isErrBecauseOfContextCancellation(ctx context.Context, err error) bool {
+	if err == nil {
+		return false
+	}
+
+	var (
+		isCancelled        = strings.Contains(err.Error(), context.Canceled.Error())
+		isDeadlineExceeded = strings.Contains(err.Error(), context.DeadlineExceeded.Error())
+		isContextDone      bool
+	)
+
+	select {
+	case <-ctx.Done():
+		isContextDone = true
+	default:
+		isContextDone = false
+	}
```

Comment:
`ctx.Err()` returns a `nil` error when it's not closed and a non-`nil` error when closed. Wouldn't be simpler to use the function exposed on the context rather than the select? I know it won't block but I struggle a bit to understand why we had a select in the first place.

```
// If Done is not yet closed, Err returns nil.
// If Done is closed, Err returns a non-nil error explaining why:
[...]
```

Reply:
Did not think about using the `.Err()` return value, thanks for noticing. I'll change that.